clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/client/AzureNetworkClient.groovy

```diff
@@ -231,7 +231,7 @@ class AzureNetworkClient extends AzureBaseClient {
       log.error("getAppGatewaysAll -> Unexpected exception ", e)
     }
 
-    result
+    result.asCollection() as Collection<AzureAppGatewayDescription>
```

Comment:
Instead of casting here, did you try changing
```
def result = []
```
to
```
Collection<AzureAppGatewayDescription> result = []
```
?

Reply:
I didn't try that, that seems more reasonable.

Reply:
Update the code.